pom.xml

```diff
@@ -737,6 +746,29 @@
         <artifactId>snakeyaml</artifactId>
         <version>${snakeyaml.version}</version>
       </dependency>
+
+      <!-- log4j -->
+      <dependency>
+        <groupId>org.slf4j</groupId>
+        <artifactId>slf4j-api</artifactId>
+        <version>${slf4j.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-slf4j-impl</artifactId>
```

Comment:
When slf4j.version is defined as 2.x, `log4j-slf4j2-impl` should be used here. So we should define this `artifactId` as a `property`, and override it in the profile where it is needed

Reply:
Done.

Reply:
> When `slf4j.version` is defined as 2.x, `log4j-slf4j2-impl` should be used here. So we should define this `artifactId` as a `property`, and override it in the profile where it is needed

It seems `flatten-maven-plugin` does not support the way you mentioned. It will cause `Duplicate dependency` error.
I've implemented it in another way. PTAL. @LuciferYang